src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs

```diff
@@ -395,10 +395,22 @@ public Task PublishUpdateAsync(IResource resource, string resourceId, Func<Custo
     /// </summary>
     private static CustomResourceSnapshot UpdateHealthStatus(IResource resource, CustomResourceSnapshot previousState)
     {
-        // A resource is also healthy if it has no health check annotations and is in the running state.
-        if (previousState.HealthStatus is not HealthStatus.Healthy && !resource.TryGetAnnotationsIncludingAncestorsOfType<HealthCheckAnnotation>(out _) && previousState.State?.Text == KnownResourceStates.Running)
+        // A resource is also healthy if it
+        // - has a null health status (wasn't set by a health check nor initial snapshot)
+        // - has no health check annotations
+        // - is in the running state
+        // - was not started with an initial snapshot
+        if (previousState.HealthStatus is null)
         {
-            return previousState with { HealthStatus = HealthStatus.Healthy };
+            var hasHealthChecks = resource.TryGetAnnotationsIncludingAncestorsOfType<HealthCheckAnnotation>(out _);
+            var hasInitialSnapshot = resource.TryGetLastAnnotation<ResourceSnapshotAnnotation>(out _);
```

Comment:
I don't think this works. Someone could add a health report with ResourceNotificationService after starting a resource.

Reply:
For example:

```cs
await notificationService.PublishUpdateAsync(
    resource,
    state => state with
    {
        State = new("Running", "success"),
        HealthReports = [new HealthReportSnapshot($"{name}_check", status, description, exception)]
    });
```

Reply:
What do you think about 
- removing this method entirely
- on the dashboard side, say a resource with a null health status is healthy if it is running with 0 health reports
- not make any assumptions other than above about the resource status - if users are updating health reports themselves, they need to handle health status too

@JamesNK

Reply:
Here is what you could do:

1. Exit out of the method if `HealthCheckAnnotation` is set. Assume that the system will manage updating health status
2. If state is not running then set HealthStatus to null. If it's not running then there is no status. This is needed for resources that start/stop. Shouldn't have a healthy status if the app isn't running.
3. If state is running then look at health reports to figure out status. No reports = healthy. A degragded report = degradged, etc.

In other words, let aspire manage the health status if `HealthCheckAnnotation` is set, otherwise the health reports on the snapshot are the source of truth.

Having both health status and health reports on the snapshot be mutable is a point of friction. Either they can go out of sync (could be unexpected) or the health reports overwrite what is manually set.